python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -748,6 +750,11 @@ def dipoleModelFunctor(x, flux, xcenPos, ycenPos, xcenNeg, ycenNeg, fluxNeg=None
         if np.any(~mask):
             weights[~mask] = 0.
 
+        # Filter out any nans, and make the weights 0.
+        nans = (~np.isfinite(z) | ~np.isfinite(weights))
```

Comment:
If you are only worried about `NaN`s, I think `np.isnan(z) | np.isnan(weights)` would be more clear.

Reply:
I guess infinities could be legal?  I've changed it.